python/pyarrow/tests/interchange/test_conversion.py

```diff
@@ -192,6 +187,34 @@ def test_pandas_roundtrip(uint, int, float, np_float):
     assert table_protocol.column_names() == result_protocol.column_names()
 
 
+@pytest.mark.pandas
+def test_roundtrip_pandas_string():
+    # See https://github.com/pandas-dev/pandas/issues/50554
+    if Version(pd.__version__) < Version("1.6"):
+        pytest.skip(" Column.size() called as a method in pandas 2.0.0")
+
+    # large string is not supported by pandas implementation
+    table = pa.table({"a": pa.array(["a", "", "c"])})
+
+    from pandas.api.interchange import (
+        from_dataframe as pandas_from_dataframe
+    )
+    pandas_df = pandas_from_dataframe(table)
+    result = pi.from_dataframe(pandas_df)
+
```

Comment:
Is there a `assert table.equals(result)` missing here (like there is in the test above)?

Reply:
Due to pandas defining `int64` offset for what is in our case normal string, not large, the dtype that is at the end of the roundtrip becomes `large_string`. Due to that, the assertion is done with pylist for the vakues and separate for dtype (first inormal string, then large string).